presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/PlanAssert.java

```diff
@@ -44,13 +44,11 @@ public static void assertPlan(Session session, Metadata metadata, StatsProvider
     {
         MatchResult matches = actual.getRoot().accept(new PlanMatchingVisitor(session, metadata, statsProvider, lookup), pattern);
         if (!matches.isMatch()) {
-            String formattedPlan = textLogicalPlan(planSanitizer.apply(actual.getRoot()), actual.getTypes(), metadata.getFunctionManager(), StatsAndCosts.empty(), session, 0);
```

Comment:
I believe the unresolved plan is for debugging iterative optimizer. It's worth keeping it still.

Reply:
However it will contain `GroupReference` nodes which are not suppose to be in the plan tree. `TranslateExpression` will not be able to run on such plan trees and thus cannot translate the `OriginalExpression`.

Reply:
`GroupReference` is for debugging purpose to check if iterative optimizer works. If someone changed `memo`, we need this to debug if the change is correct. I'm ok leaving it out for now; but just be aware of the use case.

Reply:
I will add a TODO to add it back once we have removed all `OriginalExpression`. Also, I think `memo` changes should be captured by `TestMemo` not integration tests.

Reply:
Do we have an issue tracking all the TODOs? I start to worry we might forget some of the clean ups needed as the TODO list is getting long.

Reply:
Created an issue: https://github.com/prestodb/presto/issues/13231